### ISSUE TITLE
Issue with QueuesFluentDriver

### ISSUE THREAD

**Issue by rrenna** [vapor-community/queues-database-hooks#4]
An error of `Could not send dispatched notification: server: column "payload" is of type jsonb but expression is of type bytea` is printed when used with QueuesFluentDriver.

**Comment by 0xTim** [comment]
This should probably be created on https://github.com/m-barthelemy/vapor-queues-fluent-driver, this repo has nothing to do with Fluent

**Comment by uphv** [comment]
Hi,

According to: https://github.com/vapor/postgres-nio/issues/567#event-18338365395 there is a fix to be made to the "_queue_job_completions" schema creation.

Can you please check?

Thanks!

**Comment by 0xTim** [comment]
A fix has been pushed to PostgresNIO it should now be fixed - want to try?

**Comment by uphv** [comment]
I'll try.

Thanks!

**Comment by uphv** [comment]
Hi,

Unfortunately, it doesn't work.

The error is: 

`[ WARNING ] Job failed, retrying [error: HTTPClientError.connectTimeout, job-id: 85D10ADA-D16A-4433-BDF4-CDBE0329FD78, job-name: FetchPeerJob, next-attempt: 2, queue: fetchPeer, retries-left: 1, retry-delay: 0]`

and the SQL error is:

`[ ERROR ] Job run failed [error: PSQLError(code: server, serverInfo: [sqlState: 23505, detail: Key (id)=(85D10ADA-D16A-4433-BDF4-CDBE0329FD78) already exists., file: nbtinsert.c, line: 666, message: duplicate key value violates unique constraint "_jobs_meta_pkey", routine: _bt_check_unique, localizedSeverity: ERROR, severity: ERROR, constraintName: _jobs_meta_pkey, schemaName: public, tableName: _jobs_meta], triggeredFromRequestInFile: PostgresKit/PostgresDatabase+SQL.swift, line: 81, query: PostgresQuery(sql: INSERT INTO "_jobs_meta" ("id", "queue_name", "job_name", "queued_at", "delay_until", "state", "max_retry_count", "attempts", "payload", "updated_at") VALUES ($1, $2, $3, $4, $5, 'initial', $6, $7, $8, now()), binds: [(****; TEXT; format: binary), (****; TEXT; format: binary), (****; TEXT; format: binary), (****; TIMESTAMPTZ; format: binary), (****; TIMESTAMPTZ; format: binary), (****; BIGINT; format: binary), (****; BIGINT; format: binary), (****; BYTEA; format: binary)]))]`

To me it looks like the error mechanism that should increment the "attempts" field in the "_jobs_meta" table, instead of an update it tries to insert and PK validation fails.

Perhaps there should be a check first for the existance of the PK?

Thanks!

**Comment by 0xTim** [comment]
Yes this looks like a separate issue. If you have a reproduction case, feel free to open another issue

**Comment by uphv** [comment]
Indeed a separate issue.  I'll create a new one.

**Comment by uphv** [comment]
Checked it and it doesn't look like it's fixed:

[ WARNING ] Failed to send notification [error: PSQLError(code: server, serverInfo: [sqlState: 42804, file: parse_target.c, hint: You will need to rewrite or cast the expression., line: 586, message: column "payload" is of type jsonb but expression is of type bytea, position: 236, routine: transformAssignedExpr, localizedSeverity: ERROR, severity: ERROR], triggeredFromRequestInFile: PostgresKit/PostgresDatabase+SQL.swift, line: 57, query: PostgresQuery(sql: INSERT INTO "_queue_job_completions" ("id", "jobId", "jobName", "queueName", "payload", "maxRetryCount", "delayUntil", "queuedAt", "dequeuedAt", "completedAt", "errorString", "status", "createdAt", "updatedAt") VALUES ($1, $2, $3, $4, $5, $6, NULL, $7, DEFAULT, NULL, NULL, $8, $9, $10) RETURNING "id", binds: [(****; UUID; format: binary), (****; TEXT; format: binary), (****; TEXT; format: binary), (****; TEXT; format: binary), (****; BYTEA; format: binary), (****; BIGINT; format: binary), (****; TIMESTAMPTZ; format: binary), (****; BIGINT; format: binary), (****; TIMESTAMPTZ; format: binary), (****; TIMESTAMPTZ; format: binary)])), job-id: F87AB0FB-9D4F-4E87-88AF-BB1ED904486F, job-name: ProcessInstanceJob, kind: dispatch, queue: processrawinstance, request-id: 6BB8B973-048C-46BE-B897-0C69FD4D1653]

Thanks!

**Comment by 0xTim** [comment]
What commit of PostgresNIO were you using?

**Comment by uphv** [comment]
1.26.2 latest release.

**Comment by 0xTim** [comment]
The fix hasn't been tagged yet, you need to try with `main`

**Comment by uphv** [comment]
Tried with postgres-nio  1.27.0 and I get the same:

`[ WARNING ] Failed to send notification [error: PSQLError(code: server, serverInfo: [sqlState: 42804, file: parse_target.c, hint: You will need to rewrite or cast the expression., line: 586, message: column "payload" is of type jsonb but expression is of type bytea, position: 236, routine: transformAssignedExpr, localizedSeverity: ERROR, severity: ERROR], triggeredFromRequestInFile: PostgresKit/PostgresDatabase+SQL.swift, line: 57, query: PostgresQuery(sql: INSERT INTO "_queue_job_completions" ("id", "jobId", "jobName", "queueName", "payload", "maxRetryCount", "delayUntil", "queuedAt", "dequeuedAt", "completedAt", "errorString", "status", "createdAt", "updatedAt") VALUES ($1, $2, $3, $4, $5, $6, $7, $8, DEFAULT, NULL, NULL, $9, $10, $11) RETURNING "id", binds: [(****; UUID; format: binary), (****; TEXT; format: binary), (****; TEXT; format: binary), (****; TEXT; format: binary), (****; BYTEA; format: binary), (****; BIGINT; format: binary), (****; TIMESTAMPTZ; format: binary), (****; TIMESTAMPTZ; format: binary), (****; BIGINT; format: binary), (****; TIMESTAMPTZ; format: binary), (****; TIMESTAMPTZ; format: binary)])), job-id: 070AC87B-42E7-4B66-9D17-82E273F1B35E, job-name: FetchPeerJob, kind: dispatch, queue: fetchPeer, request-id: D92C9D30-99B5-4089-8CEB-202EFEAA9DB2]`

By fix, did you mean [https://github.com/vapor/postgres-nio/pull/568](https://github.com/vapor/postgres-nio/pull/568) ?  If yes it's not working.

Thanks!
